src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java

```diff
@@ -216,7 +216,7 @@ public CaseActionsForCaseWorkerController(CaseCreationForCaseWorkerService caseC
         DefaultValues defaultValues = getPostDefaultValues(ccdRequest.getCaseDetails());
         log.info("Post Default values loaded: " + defaultValues);
         CaseData caseData = defaultValuesReaderService.getCaseData(ccdRequest.getCaseDetails().getCaseData(), defaultValues, false);
-        errors = eventValidationService.validateReceiptDate(ccdRequest.getCaseDetails().getCaseData().getReceiptDate());
+        errors = eventValidationService.validateReceiptDate(caseData);
```

Comment:
We could check the errors before actually doing any changes in the caseData.
errors = validateReceiptData
if errors then return the payload
if not then do the getPostDefaultValues and getCaseData

Reply:
noted. I will move the validation call above the defaultvalues code.

Reply:
refactored in brach : ERP-216-Date-Returned-From_Judge-Validation